parsers/CA_ON.py

```diff
@@ -257,9 +258,196 @@ def fetch_exchange(
+        NAMESPACE = "{http://www.ieso.ca/schema}"
+        ns = {"ns0": "http://www.ieso.ca/schema"}
+
+        forecast_date = xml.find(".//ns0:DeliveryDate", ns)
+
+        for forecast_ontario_demand in xml.findall(".//ns0:ForecastOntDemand", ns):
+            for demand in forecast_ontario_demand.findall(".//ns0:Demand", ns):
+                delivery_hour = demand.findtext(NAMESPACE + "DeliveryHour")
+                date_object = datetime.strptime(forecast_date.text, "%Y-%m-%d").replace(
+                    tzinfo=TIMEZONE
+                )
+                date_object = date_object.replace(hour=int(delivery_hour) - 1)
+                energy_mw = demand.findtext(NAMESPACE + "EnergyMW")
+                all_consumption_events[date_object] = energy_mw
+
+        current_date += timedelta(days=1)  # Go to next day
+
+    consumption_list = TotalConsumptionList(logger)
+    for datetime_, consumption_value in all_consumption_events.items():
+        consumption_list.append(
+            zoneKey=zone_key,
+            datetime=datetime_,
+            consumption=int(consumption_value),
+            source=SOURCE,
+            sourceType=EventSourceType.forecasted,
+        )
+    return consumption_list.to_list()
+
+
+def read_adequacy_report(root):
+    """Helper function for fetch_wind_solar_forecasts() that reads Adequacy report for a given xml root"""
+    NAMESPACE = "{http://www.ieso.ca/schema}"
+    ns = {"ns0": "http://www.ieso.ca/schema"}
+
+    # Dictionary to store extracted data
+    mixes: defaultdict[datetime, ProductionMix] = defaultdict(ProductionMix)
+    forecast_date = root.find(".//ns0:DeliveryDate", ns)
+
+    # Iterate through InternalResource elements
+    for resource in root.findall(".//ns0:InternalResource", ns):
+        mode = resource.findtext(NAMESPACE + "FuelType")
+        if mode == "Wind" or mode == "Solar":
+            for c in resource.findall("ns0:Forecasts", ns):
+                for value in c.findall("ns0:Forecast", ns):
+                    date_object = datetime.strptime(
+                        forecast_date.text, "%Y-%m-%d"
+                    ).replace(tzinfo=TIMEZONE)
+                    delivery_hour = value.findtext(NAMESPACE + "DeliveryHour")
+                    date_object = date_object.replace(hour=int(delivery_hour) - 1)
+
+                    # energy_mw = value.find("ns0:EnergyMW", ns)
+                    energy_mw = value.findtext(NAMESPACE + "EnergyMW")
+
+                    mixes[date_object].add_value(
+                        mode.lower(), None if energy_mw is None else float(energy_mw)
+                    )
+    return mixes
+
+
+def read_VGForecastSummary_report(root):
+    """Helper function for fetch_wind_solar_forecasts() that reads VGForecastsSummary report for a given xml root"""
+
+    ns = {"ns0": "http://www.ieso.ca/schema"}
+
+    # Dictionary to store extracted data
+    mixes: defaultdict[datetime, ProductionMix] = defaultdict(ProductionMix)
```

Comment:
Why mixes instead of returning ProductionLists?

It's easier to combine the production lists later on than looping over all the mixes again and the datetimes are already added.

ProductionLists can be combined with a simple `+` operator.